frontend/src/scenes/funnels/funnelLogic.ts

```diff
@@ -1069,9 +1097,32 @@ export const funnelLogic = kea<funnelLogicType>({
             actions.loadEventWithPropertyCorrelations(eventName)
         },
 
-        excludeProperty: () => {
+        excludeProperty: ({ propertyName }) => {
+            const oldCurrentTeam = teamLogic.values.currentTeam
+
+            if (!oldCurrentTeam) {
+                return
+            }
+
+            const oldCorrelationConfig = oldCurrentTeam.correlation_config
+
+            const excludedPropertyNames = (oldCorrelationConfig?.excluded_person_property_names || []).concat([
+                propertyName,
+            ])
+
+            const correlationConfig = {
+                ...oldCorrelationConfig,
+                excluded_person_property_names: excludedPropertyNames,
+            }
+
+            teamLogic.actions.updateCurrentTeam({
+                correlation_config: correlationConfig,
+            })
+
+            // NOTE: we don't wait for updateCurrentTeam to finish, we just
+            // optimistically update setPropertyNames
             actions.setPropertyNames(
-                values.propertyNames.filter((property) => !values.excludedPropertyNames.includes(property))
+                values.propertyNames.filter((property) => !excludedPropertyNames?.includes(property))
```

Comment:
Doesn't this clash with what we're already doing here: https://github.com/PostHog/posthog/pull/6715/files#diff-92e32501a6b77e6bdff549cb4f721a309c1b5fd3df14615dc6969315b66041dcR349

Reply:
checking, I added this as the tests seemed to be failing, but this probably wasn't the right fix. Think there's some action ordering issue 🤔

Reply:
I've 🤞 resolved this by listening for updateCurrentTeamSuccess instead, although I'm not totally happy.